safe-global/web-core
# feat: AB testing system

## What it solves

Partially resolves #968

## How this PR fixes it

A new AB testing system has been implemented, adding `abTest` to GTM events (if `_abTest` is currently set). `useABTesting` (which should be called in `_app.tsx`, as well as in the required places).

## How to test it

Create a dummy `AbTest` and then call `useABTesting` in `_app.tsx`. Observe the `localStorage` entry successfully created and events dispatched with `abTest`.

## Analytics changes

`abTest` string should now be present in events.

## Screenshots

![image](https://user-images.githubusercontent.com/20442784/199016659-7a6812e2-6607-4226-be40-500635dd7ca1.png)
